src/foundry/foundry.js/clientDocuments/canvasDocuments/tokenDocument.d.ts

```diff
@@ -0,0 +1,249 @@
+import { ConfiguredDocumentClass } from '../../../../types/helperTypes';
+import { AnyDocumentData } from '../../../common/abstract/data.mjs';
+import { DocumentModificationOptions } from '../../../common/abstract/document.mjs';
+import * as data from '../../../common/data/data.mjs';
+import { ActorDataSource } from '../../../common/data/data.mjs/actorData';
+
+declare global {
+  /**
+   * The client-side Token document which extends the common BaseToken model.
+   * Each Token document contains TokenData which defines its data schema.
+   *
+   * @see {@link data.TokenData}                The Token data schema
+   * @see {@link documents.Scene}               The Scene document type which contains Token embedded documents
+   * @see {@link applications.TokenConfig}      The Token configuration application
+   */
+  class TokenDocument extends CanvasDocumentMixin(foundry.documents.BaseToken) {
+    /**
+     * @param data    - Initial data provided to construct the Token document
+     *                  (default: `{}`)
+     * @param context - The parent Scene document to which this Token belongs
+     */
+    constructor(
+      data?: ConstructorParameters<ConstructorOf<foundry.documents.BaseToken>>[0],
+      context?: ConstructorParameters<ConstructorOf<foundry.documents.BaseToken>>[1]
+    );
+
+    /**
+     * A cached reference to the Actor document that this Token modifies.
+     * This may be a "synthetic" unlinked Token Actor which does not exist in the World.
+     */
+    protected _actor: InstanceType<ConfiguredDocumentClass<typeof Actor>> | null;
+
+    /**
+     * A lazily evaluated reference to the Actor this Token modifies.
+     * If actorLink is true, then the entity is the primary Actor document.
+     * Otherwise the Actor entity is a synthetic (ephemeral) document constructed using the Token's actorData.
+     */
+    get actor(): InstanceType<ConfiguredDocumentClass<typeof Actor>> | null;
+
+    /**
+     * An indicator for whether or not the current User has full control over this Token document.
+     */
+    get isOwner(): boolean;
+
+    /**
+     * A convenient reference for whether this TokenDocument is linked to the Actor it represents, or is a synthetic copy
+     */
+    get isLinked(): boolean;
+
+    /**
+     * Return a reference to a Combatant that represents this Token, if one is present in the current encounter.
+     */
+    get combatant(): InstanceType<ConfiguredDocumentClass<typeof Combatant>> | null;
+
+    /**
+     * An indicator for whether or not this Token is currently involved in the active combat encounter.
+     */
+    get inCombat(): boolean;
+
+    /**
+     * @param data    - (default: `{}`, unused)
+     * @param options - (default: `{}`, unused)
+     * @override
+     */
+    clone(
+      data?: Parameters<foundry.abstract.Document<foundry.data.TokenData>['clone']>[0],
+      options?: Parameters<foundry.abstract.Document<foundry.data.TokenData>['clone']>[1]
+    ): this;
```

Comment:
I guess we need to widen `Document.clone`. `Token.clone` does never return a Promise.

Reply:
Yeah, you are right. It should just be
```ts
  clone(
    data?: DeepPartial<Parameters<ConcreteDocumentData['_initializeSource']>[0]>,
    { save, keepId }?: { save?: boolean; keepId?: boolean }
  ): this | Promise<this>;
```